doc/tutorial.rst

```diff
@@ -89,117 +97,140 @@ We will use the following very simple Dockerfile:
+
+SquashFS:
+::
 
+  $ ch-convert hello /var/tmp/hello.sqfs
+  input:   ch-image  hello
+  output:  squash    /var/tmp/hello.sqfs
+  packing ...
+  [...]
+  done
+  $ ls -ld /var/tmp/hello.sqfs 
+  -rw-r--r-- 1 root root 83288064 Nov 15 12:07 /var/tmp/hello.sqfs
+    
+tarball:
 ::
 
-  $ sudo docker images
-  REPOSITORY  TAG      IMAGE ID      CREATED        SIZE
-  centos      8        0d120b6ccaa8  2 months ago   215MB MB
-  hello       latest   eb5f42d5bb54  5 minutes ago  235MB MB
-  $ sudo docker push  # FIXME
+  $ ch-convert hello /var/tmp/hello.tgz
+  input:   ch-image  hello
+  output:  tar       /var/tmp/hello.tgz
+  exporting ...
+  done
+  $ ls -ld /var/tmp/hello.tgz
+  -rw-rw-r-- 1 heasterday heasterday 86122450 Nov 15 15:23 /var/tmp/hello.tgz
 
-Running the image with Docker is not generally useful, because Docker's
-run-time environment is significantly different than Charliecloud's, but it
-can have value when debugging Charliecloud.
+Charliecloud can also convert images between the two formats.
 
+SquashFS to tarball:
 ::
+$ ch-convert hello.sqfs hello.tgz
+input:   tar       hello.tgz
+output:  squash    hello.sqfs
+unpacking ...
+[...]
+done
+
+tarball to SquashFS:
+::
+$ charliecloud/bin/ch-convert hello.sqfs hello.tgz
+input:   squash    hello.sqfs
+output:  tar       hello.tgz
+Parallel unsquashfs: Using 8 processors
+[...]
+done
 
-  $ sudo docker run -it hello /bin/bash
-  # ls /
-  bin   dev  hello  lib    media  opt   root  sbin  sys  usr
-  boot  etc  home   lib64  mnt    proc  run   srv   tmp  var
-  # exit
-  exit
-
-Flatten image
--------------
 
-Next, we flatten the Docker image into a tarball, which is then a plain file
-amenable to standard file manipulation commands. This tarball is placed in an
-arbitrary directory, here :code:`/var/tmp`.
+Charliecloud also supports "pushing" images from its internal storage as a
+:code:`ch-image` subcommand:
 
 ::
+    
+  $ ch-image push NAME:TAG REMOTE
 
-  $ ch-builder2tar hello /var/tmp
-  74M /var/tmp/hello.tar.gz
 
-Distribute tarball
+Distribute images
 ------------------
 
 Thus far, the workflow has taken place on the build system. The next step is
-to copy the tarball to the run system. This can use any appropriate method for
+to copy the built image to the run system. This can use any appropriate method for
 moving files: :code:`scp`, :code:`rsync`, something integrated with the
 scheduler, etc.
 
 If the build and run systems are the same, then no copy is needed. This is a
 typical use case for development and testing.
 
-Unpack tarball
+Prepping image
 --------------
 
-Charliecloud runs out of a normal directory rather than a filesystem image. In
-order to create this directory, we unpack the image tarball. This will replace
-the image directory if it already exists.
+Charliecloud can either run out of a normal directory or a mounted SquashFS
+with the latter being the recommended route. In the case of the TAR archive
```

Comment:
I read recently that "latter" and "former" can be confusing and realized I agree (similarly, a list of things followed by "respectively"). The suggested alternative is to move things closer together so you don't need such words. For example: "Charliecloud can run out of a normal directory or a mounted SquashFS (preferred)." (Though in this case IMO it's the internal SquashFS mount that's preferred, not the manual mount/unmount.)

Reply:
Excellent point, I will adjust.